### PR TITLE
Update README with release instructions

### PR DESCRIPTION
**Pull request by skaasten**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,9 +33,20 @@ Alternatively, the sources can be generated independently with:
 
     $ lein protoc
 
-## License
+Releasing
+---------
```
Comment (calvis) [inline README.md]:
```suggestion
## Releasing
```

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+:toc:
```
Comment (calvis) [inline CHANGELOG.adoc]: any particular reason for making this an adoc? i think we decided that was a failed experiment and that we'd use markdown from now on

Comment (skaasten) [reply CHANGELOG.adoc]: Nope, just based it off of [lein-deps-plus](https://github.com/circleci/lein-deps-plus/blob/main/CHANGELOG.adoc). I'll switch it to markdown.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Changelog
+
+All notable changes to this project are documented in this file.
+
+The format is based on https://keepachangelog.com/en/1.0.0/[Keep a Changelog].
```
Comment (calvis) [inline CHANGELOG.md]:
```suggestion
The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
```

File: README.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ build environments.
 
 ## Usage
 
-Put `[lein-protoc "0.5.0"]` into the `:plugins` vector of your project.clj.
+Put `[lein-protoc "0.1.0-SNAPSHOT"]` into the `:plugins` vector of your project.clj.
```
Comment (calvis) [inline README.md]: needs the group:
```suggestion
Put `[com.circleci/lein-protoc "0.1.0-SNAPSHOT"]` into the `:plugins` vector of your project.clj.
```

i think this should be versioned as `"0.6.0"`:
* as it's got all the stuff from the real lein protoc thru `0.5`. could be confusing to have a visually downgraded version
* projects shouldn't have snapshots long term, only for experimentation. this is fine for deps-plus which is typically added to the user profile, but if i was to add this to one of our libraries, i'd want a stable version for reproducibility

**Review by calvis — APPROVED** [review]
🎉

**Review by liamchzh — APPROVED** [review]

**Pull request merged**